build_tools/cmake/iree_check_test.cmake

```diff
@@ -36,6 +36,12 @@ function(iree_check_test)
     return()
   endif()
 
+  # On the host, only build check tests if the compiler is enabled.
+  # When cross compiling, assume the host tools were already built.
+  if(NOT IREE_BUILD_COMPILER AND NOT CMAKE_CROSSCOMPILING)
```

Comment:
Ok it took me a while to figure out how the comment corresponded to the if here. What if I want to use installed binaries while testing on the host? Basically, I think cross compiling doesn't have much to do with it here, so this feels like not quite the right way to check it. That said, that wasn't possible with our old cross-compiling either, so I'm fine with this workaround for now. But please tag an issue to clean this up. I think we might want something like `IREE_USE_INSTALLED_HOST_TOOLS`?

Reply:
Yeah, I think adding another cmake option makes the most sense here longer term. We could do something like set that new option on/off by default when cross compiling.

Reply:
Added way more detail to the comment.

Reply:
Thanks :-) Could you please link an issue?

Reply:
Filed and linked to https://github.com/google/iree/issues/4662